### DISCUSSION TITLE
Re-send user invitation

### DISCUSSION THREAD

**Discussion by gkielwasser** [directus/directus#8166] [Archive]
Hi,

I'm using the "inviteUser" method from the User service to handle email confirmation flow. The main issue I encounter is it can be called once per-account as this method will create the user account and send the invitation link.

Shouldn't we split this method so the invitation part can be reused without the need to create a new user account when calling it?

Currently we cannot use this method in the following cases:
- the user miss its email
- the user didn't accept fast enough the invitation (currently token expires after 7 days)

For now I don't see any means to not duplicate [this part of code](https://github.com/directus/directus/blob/d64ca14348184e69e4348c8c61bbf88ffcc9937a/api/src/services/users.ts#L307-L323)

Thanks for feedback about that

**Comment by azrikahar** [comment]
Hi, thanks for the bring this up. I do agree that there should be a mechanism to resend invites, perhaps a button showing in the user profile when their status is "invited". Though that might be tedious if there's a need for bulk resends.. 🤔

I also wonder is there a need to invalidate the previous invites (probably not?), and from my understanding this invalidation won't be possible with the existing implementation, but I could be wrong. Maybe @rijkvanzanten can help shed some light on the feasibility and/or implementation of this use case here.

**Reply by benhaynes** [reply]
Kind of ties into this Discussion too:

https://github.com/directus/directus/discussions/7394

Right now we have ephemeral in-app alerts (disappear) and Activity (events)... but no way to actually have a list of notifications that are actionable. On the admin side, I think resending an invite is just an option within `Settings > Roles > Users`. We should also add the "Status" to this interface... so it's easier to see if they are invited or active.

**Reply by rijkvanzanten** [reply]
> Hi, thanks for the bring this up. I do agree that there should be a mechanism to resend invites

I think it could be as simple as just reusing the existing user invite flow to invite the same email address again. Right now it'll prevent you from doing so with a `Email "rijk@directus.io" has already been invited` message, but we can tweak that so it'll re-send the invite email when the user's status is "invited"

**Reply by david-zacharias** [reply]
@rijkvanzanten would you accept a PR changing the behavior to allow re-invites? If yes, should refinement take place within this discussion or should I start a new one? I would be willing to contribute.

**Reply by rijkvanzanten** [reply]
@david-zacharias Lets create [a new feature request](https://github.com/directus/directus/discussions/new?category=feature-requests) about this, discuss the ideal implementation there, and then create a PR 🙂

**Reply by david-zacharias** [reply]
Thanks for guiding me, I have created [a new feature request](https://github.com/directus/directus/discussions/17856) to the best of my knowledge

**Reply by david-zacharias** [reply]
@rijkvanzanten if you find a spare minute I'd appreciate your optinion on how to solve the issue with changed role information on re-sending an invite. Details and ideas for handling this are noted in the [feature request](https://github.com/directus/directus/discussions/17856).

**Reply by rijkvanzanten** [reply]
Tricky question that one @david-zacharias! Put my notes in the feature request 👍🏻
